### DISCUSSION TITLE
Add a `.gitignore` file by default

### DISCUSSION THREAD

**Discussion by schmidsi** [graphprotocol/graph-tooling#1600] [Ideas]
I think there was a `.gitignore` file created initially. Now it's not there anymore. Is there a specific reason to not create a .gitignore file by default?

See: https://github.com/graphprotocol/graph-tooling/issues/391

**Comment by saihaj** [comment]
The question is do we recommend people committing the generated artifacts from `graph codegen` or not?

**Reply by schmidsi** [reply]
I'd say we ignore `generated` and `build`. Why would someone want to commit it? The users I saw that deploy the subgraph in a CI are able to run `graph codegen && graph build` there too.

**Reply by saihaj** [reply]
> Why would someone want to commit it?

great question. I also fall in the clan to not commit generated assets but some people prefer having the generated code in the repo to ensure consistency and avoiding a CI step (I always find it hard to convince myself)

**Reply by schmidsi** [reply]
Then I think we can agree on ignoring it. If someone disagrees, they can still edit `.gitignore` manually, no?

**Comment by saihaj** [comment]
https://github.com/graphprotocol/graph-tooling/pull/1629

**Reply by saihaj** [reply]
released [@graphprotocol/graph-cli@0.70.0](https://github.com/graphprotocol/graph-tooling/releases/tag/%40graphprotocol%2Fgraph-cli%400.70.0)
